esiclient/v1/node_port.py

```diff
@@ -0,0 +1,123 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import json
+import logging
+from osc_lib.command import command
+from osc_lib import exceptions
+from osc_lib.i18n import _
+
+
+def _get_attrs_node(parsed_args):
```

Comment:
Consider instead:

```
def _get_attrs_node(parsed_args):
    field_list = ["name", "driver", "driver_info", "resource_class"]

    attrs = {
        k: v
        for k, v in vars(parsed_args).items()
        if k in field_list and v is not None
    }

    if 'driver_info' in attrs:
        attrs['driver_info'] = param_array_to_dict(attrs['driver_info'])

    return attrs
```

And similarly for other `_get_attrs_*` functions in this change.

Reply:
Fixed it!